File: sieve_of_eratosthenes/sieve.js
function sieve(num) {
  let array = new Array(num + 1).fill(true) // create and fill an array to num
  let upperLimit = Math.sqrt(num)
  let circledPrimes = []

  // cross out multiples
  for(var i = 2; i <= upperLimit; i++) {
    if (array[i]) {
      for (var j = i * i; j < num; j += i) {
         array[j] = false;
      }
    }
  }

  // circle primes
  for (var i = 2; i < num; i++) {
    if (array[i]) {
      circledPrimes.push(i)
    }
  }

  return circledPrimes
}

console.log(sieve(100))
// sieve(100)

File: sieve_of_eratosthenes/sieve.rb
def sieve(num)
  # creating an array the size of num
  two_to_num = (2..num).to_a

  selected_prime_nums = []
  multiples = []

  two_to_num.each do |value|
    unless multiples.include?(value)
      selected_prime_nums << value  #circle primes
      multiples.concat((value..num).step(value).to_a) # cross out multiples as possible solutions
    end
  end

  selected_prime_nums
end

print sieve(100)
